### PR TITLE
[CALCITE-5698] EXTRACT from INTERVAL partially does not follow the SQL standard

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

File: babel/src/test/resources/sql/big-query.iq [hunk 1]
```diff
@@ -371,6 +371,76 @@ SELECT EXTRACT(DAY FROM DATE '2013-12-25') AS the_day;
 
 !ok
 
+SELECT EXTRACT(MONTH FROM INTERVAL -1 MONTHS) as extr;
```
Comment (zoudan) [inline babel/src/test/resources/sql/big-query.iq:374]: make sure your first commit message(the CR title) is begin with [`${jira_ID}`]

Comment (zstan) [reply babel/src/test/resources/sql/big-query.iq:374]: @zoudan thanks, all done i hope.

File: core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java [hunk 2]
```diff
@@ -2890,12 +2891,13 @@ private static class ExtractImplementor extends AbstractRexCallImplementor {
     }
   }
 
-  private static Expression mod(Expression operand, long factor) {
+  private static Expression mod(Expression operand, long factor, boolean floorMod) {
     if (factor == 1L) {
       return operand;
     } else {
-      return Expressions.call(BuiltInMethod.FLOOR_MOD.method,
```
Comment (julianhyde) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:2897]: convert '?' to 'else if' to be consistent with existing method

also revert your change removing the line-break; short lines are better

File: core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java [hunk 3]
```diff
@@ -2890,12 +2891,13 @@ private static class ExtractImplementor extends AbstractRexCallImplementor {
     }
   }
 
-  private static Expression mod(Expression operand, long factor) {
```
Comment (julianhyde) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:2893]: add javadoc explaining the purpose of the `floorMod` parameter. (If true, negative arguments will yield a negative result.)

File: core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java (hunk 2 above)
Comment (zstan) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java:2897]: thanks, done.

**Pull request closed without merging**